miden/tests/integration/exec_iters.rs

```diff
@@ -15,7 +15,7 @@ fn test_exec_iter() {
     let traces = test.execute_iter();
     let fmp = Felt::new(2u64.pow(30));
     let next_fmp = fmp + Felt::ONE;
-    let mem = vec![(1_u64, slice_to_word(&[13, 14, 15, 16]))];
+    let mem = vec![(1_u64, slice_to_word(&[16, 15, 14, 13]))];
```

Comment:
I think this change is the result of changing the order of word elements in the memory trace. While I think it is fine to have the word reversed in the trace, I do wonder if it would be too confusing for the user to see a word in stack order when looking at memory.

Basically, I think, as far as the user is concerned, they a word at address $a$ should be laid out as $[v_0, v_1, v_2, v_3]$ rather than $[v_3, v_2, v_1, v_0]$. This would also be more consistent with `mload` and `mstore` instructions which read/write the first element of the word.

I think there are a few potential ways of addressing the above:
1. Keep the word reversal contained to memory chiplet. This way, everything outside of memory chiplet works as it used. This approach is probably not ideal as the memory chiplet would need to keep reversing the words back and forth on each request.
2. Keep the words in memory chiplet in regular order, and only reverse them when the trace is generated. This would mean that the AIR constraints would remain as you have them in this PR, and as far as the processor is concerned, memory would not be stack-ordered.
3. Keep what you have in the PR now, but reverse words for the purposes of `exec_iter` only (and other user-facing components - if we have any). While this approach is probably the most efficient, I wonder if it will create confusion in the codebase.

I am leaning slightly toward the second approach - but open to other suggestions.

Reply:
As discussed, I have reverted the changes except for the update to the helper columns. The result is:

- memory order is unchanged and looks like [v0,v1,v2,v3]
- stack order is unchanged and looks like [v3,v2,v1,v0]
- the order in the helper columns now matches the stack order (e.g. [v3, v2, v1, ...], since the constraints for the stack will use the values from the helper columns